mediator/src/mappers/cht.ts

```diff
@@ -0,0 +1,142 @@
+import { getIdType, copyIdToNamedIdentifier } from '../utils/fhir';
+import { openMRSIdentifierType } from './openmrs';
+
+export const chtDocumentIdentifierType: fhir4.CodeableConcept = {
+  text: 'CHT Document ID'
+}
+
+export const chtPatientIdentifierType: fhir4.CodeableConcept = {
+  text: 'CHT Patient ID'
+}
+
+const chwVisitType: fhir4.CodeableConcept = {
+  text: "Communtiy Health Worker Visit",
+}
+
+const homeHealthEncounterClass: fhir4.CodeableConcept = {
+  text: 'HH',
+  coding : [{
+    system: "http://terminology.hl7.org/CodeSystem/v3-ActCode",
+    code: "HH"
+  }]
+}
+
+export function buildChtPatientFromFhir(fhirPatient: fhir4.Patient): any {
```

Comment:
Note: Good to have for further enhancement: 

Is this where we are converting OpenMRS patient to CHT patient? If so:

Can we implement a CHT resource mapper strategy based on deployment-specific patient types, ANC, and pregnancy forms? This strategy will create mappers with particular functions to map FHIR resources to specific sub-resource types in CHT (e.g., patient type A, patient type B), using a strategy suited for that deployment. By doing so, we can write and execute the appropriate mapping logic. This approach will also allow us to accommodate any new patient forms, such as ANC or pregnancy forms, that may differ across deployments.

```
interface PatientMappingStrategy {
  map(fhirPatient: fhir4.Patient): any;
}

function getPatientMappingStratergy {
  static getStrategy(deploymentType: string): PatientMappingStrategy {
    switch (deploymentType) {
      case 'Gandaki':
        return new PatientWithPhoneMappingStrategy();
     case 'West Africa':
        return new PatientWithNotesInFrench()
      default:
        return new PatientWithoutPhoneMappingStrategy();
    }
  }
}

class PatientWithPhoneMappingStrategy implements PatientMappingStrategy {
  map patient with phone
}

class PatientWithoutMappingStrategy implements PatientMappingStrategy {
  optionalphone
}

Usage: 
const mappingStrategy = getPatientMappingStratergy (deploymentType);

const chtPatient = mappingStrategy.map(fhirPatient);

```